Themis/Models/Authentication.swift

```diff
@@ -18,64 +18,37 @@ class Authentication: NSObject {
 
     static var shared: Authentication!
 
-    enum AuthenticationError: Error {
-        case tokenNotFound
-    }
-
-    @objc dynamic var token: String?
-    let keychain: Keychain
+    let url: URL
 
-    override init() {
-        keychain = Keychain(service: "feedback2go.auth") // should this be changed as well?
-    }
+    @objc dynamic var authenticated: Bool = false
 
-    func storeTokenInKeychain(token: String) {
-        DispatchQueue.global().async {
-            do {
-                try self.keychain
-                    .accessibility(.whenPasscodeSetThisDeviceOnly, authenticationPolicy: [.biometryAny])
-                    .set(token, key: "token")
-            } catch let error {
-                print(error.localizedDescription)
-            }
-        }
+    init(for url: URL) {
+        self.url = url
+        super.init()
     }
 
-    func getTokenFromKeychain() {
-        DispatchQueue.global().async {
-            do {
-                self.token = try self.keychain
-                    .authenticationPrompt("Authenticate to login to App")
-                    .get("token")
-            } catch let error {
-                print(error.localizedDescription)
-            }
-        }
-    }
-
-    func deleteToken() {
-        do {
-            try keychain.remove("token")
-            self.token = nil
-        } catch let error {
-            print(error.localizedDescription)
-        }
-    }
-
-    /// This Method allows the user to Authenticate. If it doesnt throw an Error the Bearer token will be set
+    /// This Method allows the user to Authenticate. If it doesnt throw an Error the token will be set as a cookie
     /// If the rememberMe flag is set the token will be valid for 30 Days (if not 30 minutes)
     func auth(username: String, password: String, rememberMe: Bool = false) async throws {
         let body = AuthBody(username: username, password: password, rememberMe: rememberMe)
         let request = Request(method: .post, path: "/api/authenticate", body: body)
-        self.token = try await RESTController.shared.sendRequest(request) { try parseAuth(data: $0) }
+        _ = try await RESTController.shared.sendRequest(request) { _ in true }
```

Comment:
I feel like this "hack" should at least be generalized into being an own function in `RESTController.swift` (or better yet, be a function the other functions there build up on. 
```suggestion
        _ = try await RESTController.shared.sendRequest(request)
```


The `RESTController` could be implemented like this:
```
    func sendRequest(_ request: Request) async throws -> Data {
        let request = try makeURLRequest(request)
        let (data, response) = try await URLSession.shared.data(for: request)
        try validate(response: response)
        return data
    }

    func sendRequest<T: Decodable>(_ request: Request) async throws -> T {
        try await sendRequest(request, decode: { data in
            try jsonDecoder.decode(T.self, from: data)
        })
    }

    func sendRequest<T>(_ request: Request, decode: (Data) throws -> T) async throws -> T {
        return try decode(await sendRequest(request))
    }
```

Reply:
Honestly don't think it's worth it to chain the most used sendRequest method three times.